### PR TITLE
HDDS-4441. Add metrics for ACL related operations.

### PR DESCRIPTION
**Pull request by ayushtkn**
## What changes were proposed in this pull request?

Add acl related metrics in OM

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4441

## How was this patch tested?

Added  UT

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java [hunk 1]
```diff
@@ -318,6 +326,42 @@ public void testKeyOps() throws IOException {
...
+      assertCounter("NumAddAcl", 1L, omMetrics);
+
+      // Test setAcl
+      ozoneManager.setAcl(volObj, acls);
+      omMetrics = getMetrics("OMMetrics");
+      assertCounter("NumSetAcl", 1L, omMetrics);
+
+      // Test removeAcl
+      ozoneManager.removeAcl(volObj, acls.get(0));
+      omMetrics = getMetrics("OMMetrics");
```
Comment (jojochuang) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java:357]: I wonder if it is really necessary to call this again and again. Doesn't look needed.

Comment (ayushtkn) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java:357]: Well, I also thought the same, it should be required only once, But my test itself fails without that. If you observe the tests above. `testVolumeOps`, `testBucketOps` and `testKeyOps` they too are also calling ``getMetrics("OMMetrics");`` before asserting.
The `getMetrics` seems to fetch the values and keep it. it doesn't seems to be updating.

Comment (jojochuang) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java:357]: ok thx for the clarification

**Review by jojochuang — APPROVED** [review]

**Comment by ayushtkn** [comment]
Thanx @jojochuang for the review!!!

**Comment by cxorm** [comment]
Thanks @ayushtkn for the contribution.

IMHO the `setAcl`, `removeAcl` and `addAcl` are write-operation, the metrics should be updated in `OMRequest` instead of `OzoneManager`. 
(For example, when we set acl of some volume, we should update the metric in `OMVolumeSetAclRequest`)

Could you be so kind as to file a jira to update it ? (Feel free to correct me if I miss something, thanks)

**Comment by ayushtkn** [comment]
Thanx @cxorm I will raise one jira for that. But I didn't actually catch what to do.
`createDirectory`, `createFile` are also write calls, for them the metrics are also updated in OM. 
I couldn't find any metrics instance itself in `OMVolumeSetAclRequest` and why we can't update the metrics in OM?

**Comment by cxorm** [comment]
> `createDirectory`, `createFile` are also write calls, for them the metrics are also updated in OM.
> I couldn't find any metrics instance itself in `OMVolumeSetAclRequest` and why we can't update the metrics in OM?

Yeah it's a good question, 
for OMHA,  actually the write-operation of OzoneManager is processed by `OMRequest` instead of `OzoneManager`.
(The write calls in `OzoneManager` is deprecated, but the code is not removed.) 

> And the metrics being updated is `OMMetrics`, So, that should track statistics of OM only?

Right, it is used to track statistics of OM (But the write calls of OM is processed by OMRequest.)

**Comment by ayushtkn** [comment]
Ohh, Means for HA, the code flow is separate. I thought `OzoneManager` like the `NamenodeRpcServer`. It should be mentioned somewhere it is only for Non-HA :-(
Well, Thanx @cxorm for the explanation. I have tried to handle this for HA, in an Addendum PR - https://github.com/apache/ozone/pull/1584

Please have a look

**Comment by bharatviswa504** [comment]
For HA/non-HA new code is being used for write requests.

**Comment by cxorm** [comment]
Thanks @bharatviswa504  , and yes I mean non-HA/ HA is using new code.

**Pull request merged**